### PR TITLE
PXB-2668 Return the exit code for each pipe when using xbcloud

### PR DESCRIPTION
**Pull request by patrickbirch**
Merged 2.4 version
Fixed conflict in source/xbcloud/xbcloud

### CODE REVIEW

File: storage/innobase/xtrabackup/doc/source/xbcloud/xbcloud.rst [hunk 1]
```diff
@@ -8,7 +8,18 @@ The purpose of |xbcloud| is to download from the cloud and upload to the cloud t
 invoked as a pipeline with |xtrabackup| to stream directly to the cloud without
 needing a local storage.
 
-|xbcloud| stores each chunk as a separate object with a name
+.. note::
+
+   In a Bash shell, the ``$?`` parameter returns the exit code from the last binary. If you use pipes the ``${PIPESTATUS[x]}`` array parameter returns the exit codes for each binary in the pipe string.
```
Comment (alina-derkach-oaza) [inline storage/innobase/xtrabackup/doc/source/xbcloud/xbcloud.rst]: If you use pipes, the ${PIPESTATUS[x]} array

File: storage/innobase/xtrabackup/doc/source/xbcloud/xbcloud.rst [hunk 2]
```diff
@@ -8,7 +8,18 @@ The purpose of |xbcloud| is to download from the cloud and upload to the cloud t
...
+   .. code-block::
+
+      $ xtrabackup --backup --stream=xbstream --target-dir=/storage/backups/ | xbcloud put [options] full_backup
+       ...
+      $ ${PIPESTATUS[x]}
+      0 0 
+
+The *xbcloud* binary stores each chunk as a separate object with a name
 ``backup_name/database/table.ibd.NNNNNNNNNNNNNNNNNNNN``, where ``NNN...`` is a
 0-padded serial number of chunk within file. Size of chunk produced by
```
Comment (alina-derkach-oaza) [inline storage/innobase/xtrabackup/doc/source/xbcloud/xbcloud.rst]: within a file.

File: storage/innobase/xtrabackup/doc/source/xbcloud/xbcloud.rst [hunk 3]
```diff
@@ -8,7 +8,18 @@ The purpose of |xbcloud| is to download from the cloud and upload to the cloud t
...
 needing a local storage.
 
-|xbcloud| stores each chunk as a separate object with a name
+.. note::
+
+   In a Bash shell, the ``$?`` parameter returns the exit code from the last binary. If you use pipes the ``${PIPESTATUS[x]}`` array parameter returns the exit codes for each binary in the pipe string. 
+
+   .. code-block::
+
+      $ xtrabackup --backup --stream=xbstream --target-dir=/storage/backups/ | xbcloud put [options] full_backup
```
Comment (altmannmarcelo) [inline storage/innobase/xtrabackup/doc/source/xbcloud/xbcloud.rst:17]: Same as 2.4 review, use a simple `true | false` to show how it works

**Review by altmannmarcelo — CHANGES_REQUESTED** [review]

File: storage/innobase/xtrabackup/doc/source/xbcloud/xbcloud.rst (hunk 2 above)
Comment (patrickbirch) [reply storage/innobase/xtrabackup/doc/source/xbcloud/xbcloud.rst]: updated

File: storage/innobase/xtrabackup/doc/source/xbcloud/xbcloud.rst (hunk 1 above)
Comment (patrickbirch) [reply storage/innobase/xtrabackup/doc/source/xbcloud/xbcloud.rst]: Added a comma

File: storage/innobase/xtrabackup/doc/source/xbcloud/xbcloud.rst (hunk 3 above)
Comment (patrickbirch) [reply storage/innobase/xtrabackup/doc/source/xbcloud/xbcloud.rst:17]: Done.

**Review by alina-derkach-oaza — APPROVED** [review]

**Review by alina-derkach-oaza — APPROVED** [review]

**Review by altmannmarcelo — APPROVED** [review]

**Pull request merged**